src/MultivariateStats.jl

```diff
@@ -301,37 +301,61 @@ function MLJBase.fit(model::LDA, ::Int, X, y)
     # check if the given one is sensible
     out_dim ≤ def_outdim || throw(ArgumentError("`out_dim` must not be larger than `min(p, nc-1)` where `p` is the dimension of `X` and `nc` is the number of classes."))
 
+    ## Estimates prior probabilities is unspecified by user.
+    if model.priors == nothing
+        priors = proportions(yplain)
+    else
+     #check if the length of priors is same as nclasses
+     length(MLJBase.classes(model.priors)) == nclasses || throw(ArgumentError("Invalid size of `priors`"))
+     priors = MLJBase.pdf.(model.priors, class_list)
+    end
+
     core_res = MS.fit(MS.MulticlassLDA, nclasses, Xm_t, Int.(yplain);
                       method=model.method,
                       outdim=out_dim,
                       regcoef=model.regcoef,
                       covestimator_within=model.cov_w,
                       covestimator_between=model.cov_b)
+    
+    ## The original projection matrix satisfies Pᵀ*Sw*P=I
+    ## scaled projection_matrix and core_res.proj by multiplying by sqrt(N - nclasses) this ensures Pᵀ*Σ*P=I
+    ## where covariance estimate Σ = Sw / (N - nclasses)
+    core_res.proj   = core_res.proj .* sqrt(N - nclasses) 
+    core_res.pmeans = core_res.pmeans .* sqrt(N - nclasses)
 
     cache     = nothing
     report    = NamedTuple{}()
-    fitresult = (core_res, class_list)
+    fitresult = (core_res, class_list, priors)
 
     return fitresult, cache, report
 end
 
-function MLJBase.fitted_params(::LDA, (core_res, class_list))
+function MLJBase.fitted_params(::LDA, (core_res, class_list, priors))
     return (class_means       = MS.classmeans(core_res),
-            projection_matrix = MS.projection(core_res))
+            projection_matrix = MS.projection(core_res),
+            priors            = priors)
 end
 
-function MLJBase.predict(m::LDA, (core_res, class_list), Xnew)
+function MLJBase.predict(m::LDA, (core_res, class_list, priors), Xnew)
     # projection of Xnew XWt is n x o  where o = number of out dims
     XWt = MLJBase.matrix(Xnew) * core_res.proj
     # centroids in the transformed space, nc x o
     centroids = permutedims(core_res.pmeans)
 
+    N  = core_res.stats.tweight # N is the Number of training examples
+    nclasses = length(class_list) 
+
     # compute the distances in the transformed space between pairs of rows
-    # the probability matrix is `n x nc` and normalised accross rows
-    P = pairwise(m.dist, XWt, centroids, dims=1)
-    # apply a softmax transformation
-    P .-= maximum(P, dims=2)
-    P  .= exp.(-P)
+    # The discriminant matrix `P` is of dimension `n x nc`
+    #  P[i,k] = -0.5*(xᵢ −  µₖ)ᵀΣ⁻¹(xᵢ −  µₖ) + log(priorsₖ) and Σ⁻¹ = I due to the nature of the projection_matrix
+
+    P = pairwise(SqEuclidean(), XWt, centroids, dims=1)
+    P .*= -0.5
```

Comment:
remove this line, it's not necessary

Reply:
it is following the formula in the comment in the code

Reply:
ok it's fine I guess let's leave it as you put it; my reasoning is as follows: this is tied with the SqEuclidean which is tied with a specific distance; I don't really like this in that I still  think it should be possible to use another distance in there; but  that distance would need to be weighed by some other scale; 

Eg for instance if you use cosine or absolute value distance; 

On the other hand I get what you're trying  to do with Bayes rule. 

so we could either

1. offer a standard bayesian path; which would force the SqEuclidean and allow all that you were suggesting
2. offer a non-standard  path where people can  specify their own distance and potentially a weighing parameter for their distance

what do  you think?

Reply:
I guess there could be `LDA` (basically what's already there) and `BayesianLDA` (something similar but that implements things as you've done them, fixing the metric and using a gaussian model)